queue/driver_mongo.go

```diff
@@ -677,24 +677,30 @@ func (d *mongoDriver) GetAllAttempts(ctx context.Context, id string) ([]amboy.Jo
 }
 
 func (d *mongoDriver) Put(ctx context.Context, j amboy.Job) error {
-	ji, err := registry.MakeJobInterchange(j, d.opts.Format)
-	if err != nil {
-		return errors.Wrap(err, "converting in-memory job to interchange job")
-	}
-
-	ji.Scopes = j.EnqueueScopes()
+	return d.PutMany(ctx, []amboy.Job{j})
+}
 
-	d.addMetadata(ji)
+func (d *mongoDriver) PutMany(ctx context.Context, jobs []amboy.Job) error {
+	var jobInterchanges []any
+	for _, j := range jobs {
+		ji, err := registry.MakeJobInterchange(j, d.opts.Format)
+		if err != nil {
+			return errors.Wrap(err, "converting in-memory job to interchange job")
+		}
+		ji.Scopes = j.EnqueueScopes()
+		d.addMetadata(ji)
+		jobInterchanges = append(jobInterchanges, ji)
+	}
 
-	if _, err = d.getCollection().InsertOne(ctx, ji); err != nil {
+	if _, err := d.getCollection().InsertMany(ctx, jobInterchanges, options.InsertMany().SetOrdered(false)); err != nil {
 		if d.isMongoDupScope(err) {
-			return amboy.NewDuplicateJobScopeErrorf("job scopes '%s' conflict", j.Scopes())
+			return amboy.NewDuplicateJobScopeErrorf("job scopes conflict")
```

Comment:
Can we include the subset of IDs of the jobs that produced a scope conflict error? If not, can we list out all the job IDs?

Reply:
Opted to throw in all the job ids because on the premise that a duplicate key error could show up in a `WriteConcernError` or a `CommandError` we wouldn't know which job caused the error.